manager/logbroker/broker.go

```diff
@@ -54,6 +60,38 @@ func validateSelector(selector *api.LogSelector) error {
 	return nil
 }
 
+func (lb *LogBroker) registerSubscription(subscription *api.SubscriptionMessage) {
+	lb.mu.Lock()
+	defer lb.mu.Unlock()
```

Comment:
Probably better to release the lock before calling `Publish`.

I might change the name of the mutex to make it clear it protects `registeredSubscriptions` specifically.